java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -680,3 +678,14 @@ private MethodAccess callReturningSameType(Expr ref) {
   ref = result.getQualifier() and
   result.getMethod().getReturnType() = ref.getType()
 }
+
+private RefType entrypointType() {
+  result = any(RemoteFlowSource s | exists(Parameter p | p = s.asParameter())).getType() or
```

Comment:
This can be simplified to
```
  result = any(RemoteFlowSource s | exists(s.asParameter())).getType() or
```
or perhaps phrased slightly cleaner as
```suggestion
  result = any(RemoteFlowSource s | s instanceof DataFlow::ExplicitParameterNode).getType() or
```